core/src/main/java/com/oath/oak/OakDetachedBuffer.java

```diff
@@ -1,5 +1,14 @@
 package com.oath.oak;
 
+/**
+ * This detached buffer allows reuse of the same object and is used in:
+ *   (1) Oak's stream iterators, where we reuse the same detached buffer, i.e., it refers to different internal
+ *       buffers as we iterate the map.
+ *   (2) normal iterations without reusing.
+ *
+ * Unlike other ephemeral objects, even if this references a value, it does not have to acquire a read lock
```

Comment:
I don't understand this comment it must be wrong or at least not the comment on the public interface. "it does not have to acquire a read lock"? "it can only be used without other concurrent writes in the background"??

Reply:
This is a misplaced comment from the old DetachedKeyBuffer (I think).
But it is true for the case of stream-iterators (currently), albeit not specified in the manual.
I added the following comment to the "unsynced" detached buffer instead:
```
 * It does not have to acquire a read lock before each access.
 * Thus, it can only be used without other concurrent writes in the background.
```